# bronnen
* McInnes, I. B., & Schett, G. (2011). The pathogenesis of rheumatoid arthritis. New England Journal of Medicine, 365(23), 2205–2219. https://doi.org/10.1056/NEJMra1004965

* Smolen, J. S., Aletaha, D., McInnes, I. B. (2016). Rheumatoid arthritis. The Lancet, 388(10055), 2023–2038. https://doi.org/10.1016/S0140-6736(16)30173-8

* Liao, Y., Smyth, G. K., & Shi, W. (2019). The R package Rsubread is easier, faster, cheaper and better for alignment and quantification of RNA sequencing reads. Nucleic Acids Research, 47(8), e47. https://doi.org/10.1093/nar/gkz114

* Morgan, M., Obenchain, V., Langfelder, P., & Martin, M. (2023). Rsamtools: Binary alignment (BAM), FASTA, variant call (BCF), and tabix file import. Bioconductor.

* Love, M. I., Huber, W., & Anders, S. (2014). Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2. Genome Biology, 15(12), 550. https://doi.org/10.1186/s13059-014-0550-8

* Young, M. D., Wakefield, M. J., Smyth, G. K., & Oshlack, A. (2010). Gene ontology analysis for RNA-seq: accounting for selection bias. Genome Biology, 11(2), R14. https://doi.org/10.1186/gb-2010-11-2-r14

* Luo, W., & Brouwer, C. (2013). Pathview: an R/Bioconductor package for pathway-based data integration and visualization. Bioinformatics, 29(14), 1830–1831. https://doi.org/10.1093/bioinformatics/btt285